runtime/src/chpl-linefile-support.c

```diff
@@ -26,32 +26,33 @@
 #include "chpl-linefile-support.h"
 
 // chpl_filenameTable is defined in chplcgfns.h
-
-static int compare(const void *a, const void *b) {
-  return strcmp((c_string)a, *(c_string *)b);
-}
-
-uint64_t chpl_findFilenameIdx(c_string name) {
-  c_string *result;
-
-  if (name == NULL) {
-    // We will occasionally get a NULL passed in as the filename, just use
-    // <internal> instead in that case
-    name = "<internal>";
+c_string chpl_lookupFilename(const int32_t idx) {
+  if (idx < 0) {
+    switch (idx) {
+    case CHPL_FILE_IDX_COMMAND_LINE:
+      return "(command-line)";
+    case CHPL_FILE_IDX_COMMAND_LINE_ARG:
+      return "<command-line arg>";
+    case CHPL_FILE_IDX_FORK_LARGE:
+      return "fork large";
+    case CHPL_FILE_IDX_MAIN_PROGRAM:
+      return "main program";
+    case CHPL_FILE_IDX_UNKNOWN:
+      return "<unknown>";
+    case CHPL_FILE_IDX_IDLE_TASK:
+      return "|idle|";
+    case CHPL_FILE_IDX_COMM_TASK:
+      return "<comm task>";
+    case CHPL_FILE_IDX_MAIN_TASK:
+      return "<main task>";
+    default: {
+      assert(false);
+      return "";
+    }
+    }
+  } else {
+    assert(idx < chpl_filenameTableSize);
```

Comment:
The thread local buffer could potentially change between creation and use, is that OK?